### PR TITLE
Update HA Addon to use image from libretiny-eu

### PR DESCRIPTION
**Pull request by catalin2402**
To update the Home Assistant addon images, checkout this branch, update it with the mainstream and run these commands

`docker buildx create --use desktop-linux`

`docker buildx build --build-arg BASEIMGTYPE=hassio --build-arg BUILD_VERSION=latest --file docker/Dockerfile --platform linux/amd64,linux/aarch64,linux/arm/v7 --target hassio --tag ghcr.io/libretiny-eu/libretiny-esphome-hassio:latest . --push`

`docker buildx build --build-arg BASEIMGTYPE=docker --build-arg BUILD_VERSION=latest --file docker/Dockerfile --platform linux/amd64,linux/aarch64,linux/arm/v7 --target docker --tag ghcr.io/libretiny-eu/libretiny-esphome-docker:latest . --push`

### CODE REVIEW

**Comment by kuba2k2** [comment]
Why are the `libc6` hacks still needed? I thought upgrading to GCC 10.3.1 (which has all native binaries available on PIO registry) would fix the issue (didn't it?).

Doesn't the problem only occur now for people building an armv7 container on an aarch64 machine? (or the equivalent for x86)

**Comment by catalin2402** [comment]
Kept them from the old files of the addon :), I will try to remove them and see if everything is compiling fine on x64, arm64 and armv7

**Comment by kuba2k2** [comment]
Yeah, because if someone is compiling 32bit on a 64bit host, it's usually not right and we probably don't need to support this, which should make the images smaller too.

**Comment by catalin2402** [comment]
I think the difference is something like 10MB, I will test it tomorrow

**Comment by catalin2402** [comment]
Hey, I removed libc6 dependencies, I added the changes from https://github.com/esphome/esphome/commit/57e909e7909b5a11024fe630c506ba166243a68c and now the image is only 300 MB as normal :)

File: docker/Dockerfile [hunk 1]
```diff
@@ -152,4 +152,4 @@ COPY requirements_test.txt /
 RUN pip3 install --no-cache-dir -r /requirements_test.txt
 
 VOLUME ["/esphome"]
-WORKDIR /esphome
+WORKDIR /esphome
```
Comment (kuba2k2) [inline docker/Dockerfile]:
```suggestion
WORKDIR /esphome

```

File: script/setup [hunk 2]
```diff
@@ -15,4 +15,4 @@ pip3 install --no-use-pep517 -e .
 
 pre-commit install
 
-script/platformio_install_deps.py platformio.ini
+script/platformio_install_deps.py platformio.ini --libraries --tools --platforms
```
Comment (kuba2k2) [inline script/setup]:
```suggestion
script/platformio_install_deps.py platformio.ini --libraries --tools --platforms

```

File: script/platformio_install_deps.py [hunk 3]
```diff
@@ -45,4 +55,4 @@
             tools.append("-t")
             tools.append(tool)
 
-subprocess.check_call(["platformio", "pkg", "install", "-g", *libs, *platforms, *tools])
+subprocess.check_call(["platformio", "pkg", "install", "-g", *libs, *platforms, *tools])
```
Comment (kuba2k2) [inline script/platformio_install_deps.py]:
```suggestion
subprocess.check_call(["platformio", "pkg", "install", "-g", *libs, *platforms, *tools])

```

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

**Pull request merged**